backtracking/remove_invalid_parenthese.py

```diff
@@ -0,0 +1,120 @@
+def remove_invalid_parentheses(input_string: str) -> list[str]:
+    """
+    Removes the minimum number of invalid parentheses
+    to make the input string valid.
+
+    Args:
+        input_string (str): The input string containing parentheses.
+
+    Returns:
+        List[str]: A list of valid strings after removing
+        the minimum number of parentheses.
+
+    Examples:
+        >>> remove_invalid_parentheses(")(a)())")
+        ['(a())', '(a)()']
+
+        >>> remove_invalid_parentheses("()())()")
+        ['(())()', '()()()']
+
+        >>> remove_invalid_parentheses(")(")
+        ['']
+    """
+
+    def is_valid(next_string: str) -> bool:
```

Comment:
We discourage nested functions because they don't get doctested

Reply:
its dependent function its not a main function which should be doctested

Reply:
Please put the `is_valid()` doctests into the docstring of `remove_invalid_parentheses()`

Reply:
ohk sure